build_runner/test/package_graph/package_graph_test.dart

```diff
@@ -217,6 +217,96 @@ void main() {
     });
   });
 
+  group('orderedPackageCycles', () {
+    test('with two sub trees', () {
+      var a = new PackageNode('a', '1.0.0', PackageDependencyType.path, null);
```

Comment:
It might be worth adding some sort of utility to create these package structures in a more visual way such as a map?

For instance this could look like:

```dart
var graph = createPackageGraph(/*name of root node*/ 'a', {
  'a': {
    'dependencies': ['left1', 'left2'],
  },
  'left1': {
    'dependencies': ['left2'],
  },
  'left2': {
    'dependencies': [],
  },
  'right1': {
    'dependencies': ['right2'],
  },
  'right2': {
    'dependencies': [],
  },
});
```

Or we could provide in memory pubspecs or something, idk just a thought :)

Reply:
definitely worth adding later. For now this was copy/pasted from our existing tests of `orderedDependencies`